finagle-core/src/test/scala/com/twitter/finagle/loadbalancer/TrafficDistributorTest.scala

```diff
@@ -470,4 +471,20 @@ class TrafficDistributorTest extends FunSuite {
     Await.ready(client.close())
     intercept[ServiceClosedException] { Await.result(client("x")) }
   }
+
+  test("transition to empty balancer on empty address state")(new Ctx{
+    val init: Activity.State[Set[Address]] = Activity.Pending
+    val dest = Var(init)
+    val dist = newDist(dest)
+
+    dest() = Activity.Ok(Set((1, 2.0),(2, 1.0),(3, 2.0)).map(x => WeightedAddress(Address(x._1), x._2)))
+    val bal0 = Await.result(dist())
+    assert(balancers.size == 2)
+
+    dest() = Activity.Ok(Set.empty[Address])
+    val bal1 = intercept[Exception] {Await.result(dist())}
+    assert(balancers.size == 3)
```

Comment:
Are we sure this is getting rid of (or calling `close` on) the empty balancer? Can we validate this by adding a hook to the `Balancer` close method in `Ctx`?

Reply:
My manual tests of this branch, or at least looking at the `Balancers` in `loadbalancers.json`, indicated that the `Balancers` were being closed. but I can add your suggestion to verify.